lib/index.js

```diff
@@ -11,65 +10,51 @@ var isodate = require('@segment/isodate');
 module.exports = traverse;
 
 /**
- * Traverse an object or array, and return a clone with all ISO strings parsed
- * into Date objects.
+ * Recursively traverse an object or array, and convert
+ * all ISO date strings parse into Date objects.
  *
- * @param {Object} obj
+ * @param {Object} input - object, array, or string to convert
+ * @param {Boolean} strict - only convert strings with year, month, and date
  * @return {Object}
  */
 
 function traverse(input, strict) {
-  if (strict === undefined) strict = true;
-
-  if (type(input) === 'object') return object(input, strict);
-  if (type(input) === 'array') return array(input, strict);
+  if (strict === undefined)
+    strict = true;
+  if (type(input) === 'object') {
+    return traverseObject(input, strict);
+  } else if (type(input) === 'array') {
+    return traverseArray(input, strict);
+  } else if (isodate.is(input, strict)) {
+    return isodate.parse(input);
+  }
```

Comment:
I think a switch case on the value of type(input) where the default case contains an if (isodate.is(...)) would be cleaner here but it's just a matter of stylistic preference

Reply:
I'm not a huge fan of `switch` personally so gonna skip :)